base_delivery_carrier_files/stock.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Guewen Baconnier
+#    Copyright 2012 Camptocamp SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api
+
+
+class stock_picking(models.Model):
+    _inherit = "stock.picking"
+
+    @api.multi
+    def generate_carrier_files(self, auto=True,
+                               recreate=False):
+        """
+        Generates all the files for a list of pickings according to
+        their configuration carrier file.
+        Does nothing on pickings without carrier or without
+        carrier file configuration.
+        Generate files only for outgoing pickings.
+
+        :param list ids: list of ids of pickings for which we need a file
+        :param auto: specify if we call the method from an automatic action
+                     (on process on picking as instance)
+                     or called manually from the wizard. When auto is True,
+                     only the carrier files set as "auto_export"
+                     are exported
+        :return: True if successful
+        """
+        carrier_file_ids = {}
+        for picking in self:
+            if picking.picking_type_id.code != 'outgoing':
+                continue
+            if not recreate and picking.carrier_file_generated:
+                continue
+            carrier = picking.carrier_id
+            if not carrier or not carrier.carrier_file_id:
```

Comment:
no need to check carrier.
If `carrier` is Null, `carrier.carrier_file_id` will be Null too.